### PR TITLE
Export user emails and URNs/UIDs to db

### PR DESCRIPTION
**Pull request by david-mears-dfe**
## Jira ticket URL

First slice of:
https://dfedigital.atlassian.net/browse/TEVA-815

## Changes in this PR:

- In the interests of keeping PRs small, here is the first slice of the magic-sign-in-link authentication fallback: a task for filling the Users table with data from DSI. URNs and UIDs will be used for 'associating' (not in the database but informally associating) users with schools/organisations.

## Next steps:

- [ ] Front and back end authentication-by-magic-email-link pathway (part done, not in this PR)

### CODE REVIEW

File: spec/fixtures/dfe_sign_in_service_users_response_page_2.json [hunk 1]
```diff
@@ -1,13 +1,110 @@
 {
-  "users": [{
+  "users": [
+    {
     "approvedAt": "2019-06-19T15:09:58.683Z",
     "updatedAt": "2019-06-19T15:09:58.683Z",
     "organisation": {
-      "id": "13F20E54-79EA-4146-8E39-18197576F024",
-      "name": "Department for Education",
+      "id": "meaningless string",
```
Comment (cesidio) [inline spec/fixtures/dfe_sign_in_service_users_response_page_2.json:7]: Why don't you keep it in the UUID format?

File: spec/fixtures/dfe_sign_in_service_users_empty_response.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "users": [],
+    "numberOfRecords": 0,
+    "page": 1,
+    "numberOfPages": 1
+}
+
```
Comment (cesidio) [inline spec/fixtures/dfe_sign_in_service_users_empty_response.json:7]: Some extra chars at the end of the line. Better to fix it

**Comment by david-mears-dfe** [comment]
Before merging, I'll test run the task on a pre-prod environment.

**Comment by david-mears-dfe** [comment]
It didn't work last night, seemingly because of me forgetting to run the new migrations.

`rails db:migrate:status` gives:
```
   up     20200211085235  Create documents
   up     20200211085721  Add vacancy to documents
   up     20200213152822  Add gias data to column in schools
   up     20200309172258  Remove weekly hours from vacancies
   up     20200319124649  ********** NO FILE **********
   up     20200319165536  Add job role to vacancies
   up     20200325121727  Add salary to vacancies
   up     20200327125528  Drop pay scales
   up     20200401125329  Add completed step to vacancies
   up     20200405081133  Change job description to job summary
   up     20200405090140  Add about school to vacancies
   up     20200406102653  Change job summary to nullable column
   up     20200514084513  Remove minimum salary db contraint
   up     20200519151545  Add state to vacancies
   up     20200521075215  Add subjects to vacancies
  down    20200521171155  Add dsi data to users
```

```
NoMethodError (undefined method `dsi_data' for #<User:0x00005556f8574690>)
irb(main):009:0>
irb(main):010:0> User
=> User(id: uuid, oid: string, accepted_terms_at: datetime, email: string)
```

**Comment by david-mears-dfe** [comment]
I realised the database sync scheduled for 2 am would overwrite any effect of my task, on staging.

**Comment by david-mears-dfe** [comment]
It seems not to have worked tonight either, and I'm checking before the database sync.

```
irb(main):002:0> User # the migration certainly took place
=> User(id: uuid, oid: string, accepted_terms_at: datetime, email: string, dsi_data: jsonb)
irb(main):005:0> User.all.where(email: nil).size # but the task didn't run or it failed
=> 5550
```

**Comment by david-mears-dfe** [comment]
No activity on papertrail, so maybe something is up with the task scheduling.

**Comment by david-mears-dfe** [comment]
The task ran successfully with `UpdateDfeSignInUsers.new.run!` on staging, so I am merging this and creating a ticket to check the scheduling.

**Pull request merged**